Add OptimizeImplicitlyTriggeredBuild property (#26245)
dotnet/docs: docs/core/project-sdk/msbuild-props.md

> [!IMPORTANT]
> If you set the `GenerateRequiresPreviewFeaturesAttribute` property to `False`, you must be certain to decorate all public APIs that rely on preview features with <xref:System.Runtime.Versioning.RequiresPreviewFeaturesAttribute>.

### OptimizeImplicitlyTriggeredBuild

> To speed up the build time, builds that are implicitly triggered by Visual Studio skip code analysis, including nullable analysis. These implicit builds occur when you run tests or start debugging in Visual Studio. However, builds are optimized in this manner only when `TreatWarningsAsErrors` is not `true`. If you have `TreatWarningsAsErrors` set to `true` but you still want implicitly triggered builds to be optimized, you can set `OptimizeImplicitlyTriggeredBuild` to `True`. To turn off build optimization for implicitly triggered builds, set `OptimizeImplicitlyTriggeredBuild` to `False`.

Reviewer: NOTE: The feature is currently implemented only for "Run/Debug Tests" commands. We plan to implement it for "Start Debugging" and "Start Without Debugging" in Dev17.1

Author: ```suggestion
To speed up the build time, builds that are implicitly triggered by Visual Studio skip code analysis, including nullable analysis. Visual Studio triggers an implicit build when you run tests, for example. However, builds are optimized in this manner only when `TreatWarningsAsErrors` is not `true`. If you have `TreatWarningsAsErrors` set to `true` but you still want implicitly triggered builds to be optimized, you can set `OptimizeImplicitlyTriggeredBuild` to `True`. To turn off build optimization for implicitly triggered builds, set `OptimizeImplicitlyTriggeredBuild` to `False`.
```
